### PR TITLE
[Shrine] New Shrine with expanding cart bottom sheet from experimenal

### PR DESCRIPTION
**Pull request by willlarche**
Written by @michdud

### CODE REVIEW

File: shrine/lib/app.dart [hunk 1]
```diff
@@ -19,23 +19,43 @@ import 'category_menu_page.dart';
...
 class ShrineApp extends StatefulWidget {
   @override
   _ShrineAppState createState() => _ShrineAppState();
 }
 
-class _ShrineAppState extends State<ShrineApp> {
+class _ShrineAppState extends State<ShrineApp>
+    with SingleTickerProviderStateMixin {
+  // Controller to coordinate both the opening/closing of backdrop and sliding
+  // of expanding bottom sheet
```
Comment (RedBrogdon) [inline shrine/lib/app.dart]: nit: could use a period on the end of the sentence.

File: shrine/lib/backdrop.dart [hunk 2]
```diff
@@ -74,7 +76,10 @@ class _BackdropTitle extends AnimatedWidget {
 
   @override
   Widget build(BuildContext context) {
-    final Animation<double> animation = this.listenable;
+    final Animation<double> animation = CurvedAnimation(
+      parent: this.listenable,
+      curve: Interval(0.0, 0.78),
```
Comment (RedBrogdon) [inline shrine/lib/backdrop.dart]: Is there a reason 0.78 doesn't get a constant like the ones in lines 20-23?

File: shrine/lib/backdrop.dart [hunk 3]
```diff
@@ -193,27 +198,88 @@ class _BackdropState extends State<Backdrop>
   }
 
   void _toggleBackdropLayerVisibility() {
-    _controller.fling(
-        velocity: _frontLayerVisible ? -_kFlingVelocity : _kFlingVelocity);
+    // Call setState here to update layerAnimation update if that's necessary
```
Comment (RedBrogdon) [inline shrine/lib/backdrop.dart]: Not sure what "update layerAnimation update" means here.

File: shrine/lib/category_menu_page.dart [hunk 4]
```diff
@@ -21,45 +21,51 @@ import 'model/product.dart';
...
-                    ),
-                  ),
+        onTap: () {
+          model.setCategory(category);
+          if (onCategoryTap != null)
+            onCategoryTap();
+        },
+        child: model.selectedCategory == category
+            ? Column(
+          children: <Widget>[
```
Comment (RedBrogdon) [inline shrine/lib/category_menu_page.dart]: Did dart format set the indents this way (lines 44-58, 60-66)?

File: shrine/lib/supplemental/product_card.dart [hunk 5]
```diff
@@ -42,44 +42,49 @@ class ProductCard extends StatelessWidget {
...
-                  ));
-            },
-            child: child,
-          ),
-      child: Column(
-        mainAxisAlignment: MainAxisAlignment.center,
-        crossAxisAlignment: CrossAxisAlignment.center,
+        onTap: () {
+          model.addProductToCart(product.id);
+          // TODO: Add Snackbar
```
Comment (RedBrogdon) [inline shrine/lib/supplemental/product_card.dart]: Is the snackbar coming back in another PR? That was my addition, so if it doesn't fit your design, you can leave it out.

File: shrine/lib/app.dart (hunk 1 above)
Comment (willlarche) [reply shrine/lib/app.dart]: Done.

File: shrine/lib/backdrop.dart (hunk 2 above)
Comment (willlarche) [reply shrine/lib/backdrop.dart]: Usually avoid constants in the educational code but they helped make the emphasized easing function easier to read.

File: shrine/lib/backdrop.dart (hunk 3 above)
Comment (willlarche) [reply shrine/lib/backdrop.dart]: Me neither.

File: shrine/lib/category_menu_page.dart (hunk 4 above)
Comment (willlarche) [reply shrine/lib/category_menu_page.dart]: No but happy to run it.

File: shrine/lib/backdrop.dart (hunk 3 above)
Comment (willlarche) [reply shrine/lib/backdrop.dart]: Done.

File: shrine/lib/supplemental/product_card.dart (hunk 5 above)
Comment (willlarche) [reply shrine/lib/supplemental/product_card.dart]: Removed.

**Comment by willlarche** [comment]
@RedBrogdon PTAL

File: shrine/lib/backdrop.dart (hunk 3 above)
Comment (michdud) [reply shrine/lib/backdrop.dart]: Typo, sorry!

**Comment by RedBrogdon** [comment]
LGTM

**Pull request merged**